### PR TITLE
Update CSM Authorization to reflect the correct version

### PR DESCRIPTION
**Pull request by hoppea2**
# Description

The artifacts created for the deployment of CSM Authorization still reflected tech-preview versions. In preparation for the release, updated install and deployment scripts to accurately reflect the next released version. In the process increased code coverage on some packages and added packages with lower coverage to skip-list. This will be addressed in a future release.

PASS: 14/14
go test -count=1 -cover -race -timeout 30s -short ./...
?       karavi-authorization/cmd/karavictl      [no test files]
ok      karavi-authorization/cmd/karavictl/cmd  17.515s coverage: 65.9% of statements
ok      karavi-authorization/cmd/proxy-server   0.065s  coverage: 7.0% of statements
?       karavi-authorization/cmd/sidecar-proxy  [no test files]
ok      karavi-authorization/cmd/tenant-service 0.042s  coverage: 9.4% of statements
ok      karavi-authorization/deploy     0.093s  coverage: 86.3% of statements
?       karavi-authorization/internal/decision  [no test files]
ok      karavi-authorization/internal/powerflex 9.078s  coverage: 92.6% of statements
ok      karavi-authorization/internal/proxy     7.589s  coverage: 68.6% of statements
ok      karavi-authorization/internal/quota     0.331s  coverage: 90.1% of statements
ok      karavi-authorization/internal/roles     0.037s  coverage: 93.2% of statements
ok      karavi-authorization/internal/tenantsvc 2.539s  coverage: 83.1% of statements
ok      karavi-authorization/internal/token     0.034s  coverage: 90.9% of statements
ok      karavi-authorization/internal/token/jwx 0.035s  coverage: 71.7% of statements
ok      karavi-authorization/internal/web       0.034s  coverage: 23.1% of statements
?       karavi-authorization/pb [no test files]

# Issues

| Issue ID |
| #139  |
|          |

# Checklist:

- [x] I have performed a self-review of my own changes.
- [x] Tested installation and deployment including sidecar injection and test cases

### CODE REVIEW

File: deploy/install.go [hunk 1]
```diff
@@ -97,8 +97,8 @@ const (
 	certConfigManifest           = "signed-cert.yaml"
 	bundleTarPath                = "dist/karavi-airgap-install.tar.gz"
 	karavictl                    = "karavictl"
-	sidecarImageTar              = "sidecar-proxy-latest.tar"
-	sidecarDockerImage           = "sidecar-proxy:latest"
+	sidecarImageTar              = "sidecar-proxy-1.0.0.tar"
```
Comment (gallacher) [inline deploy/install.go:100]: Can we pull this out of code so we don't need to change it every time there is a new release?

File: docs/GETTING_STARTED_GUIDE.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ A Storage Administrator can execute the installer or rpm package as a root user
 ```
```
Comment (gallacher) [inline docs/GETTING_STARTED_GUIDE.md:65]: No changes required here.  Content has moved to the common doc site so please make changes there instead

File: docs/GETTING_STARTED_GUIDE.md [hunk 3]
```diff
@@ -65,7 +65,7 @@ A Storage Administrator can execute the installer or rpm package as a root user
...
 {
   "web": {
-    "sidecarproxyaddr": "docker_registry/sidecar-proxy:latest",
+    "sidecarproxyaddr": "docker_registry/sidecar-proxy:1.0.0",
```
Comment (gallacher) [inline docs/GETTING_STARTED_GUIDE.md:68]: No changes required here.  Content has moved to the common doc site so please make changes there instead

File: docs/GETTING_STARTED_GUIDE.md [hunk 4]
```diff
@@ -117,7 +117,7 @@ Karavi Authorization has a subset of configuration parameters that can be update
 
 | Parameter | Type | Default | Description |
 | --------- | ---- | ------- | ----------- |
-| web.sidecarproxyaddr | String |"127.0.0.1:5000/sidecar-proxy:latest" |Registry address of the Karavi Authorization sidecar-proxy |
+| web.sidecarproxyaddr | String |"127.0.0.1:5000/sidecar-proxy:1.0.0" |Registry address of the Karavi Authorization sidecar-proxy |
```
Comment (gallacher) [inline docs/GETTING_STARTED_GUIDE.md:120]: No changes required here.  Content has moved to the common doc site so please make changes there instead

File: docs/KARAVICTL_GUIDE.md [hunk 5]
```diff
@@ -114,15 +114,15 @@ karavictl inject [flags]
 Inject into an existing vxflexos CSI driver
 ```
 kubectl get secrets,deployments,daemonsets -n vxflexos -o yaml \
-   | karavictl inject --image-addr [IMAGE_REPO]:5000/sidecar-proxy:latest --proxy-host [PROXY_HOST_IP] \
+   | karavictl inject --image-addr [IMAGE_REPO]:5000/sidecar-proxy:1.0.0 --proxy-host [PROXY_HOST_IP] \
```
Comment (gallacher) [inline docs/KARAVICTL_GUIDE.md:117]: No changes required here.  Content has moved to the common doc site so please make changes there instead

File: internal/web/client_install_handler.go [hunk 6]
```diff
@@ -8,7 +8,7 @@ import (
 
 // DefaultSidecarProxyAddr is the default location where a client can
 // download the sidecar proxy container image.
-var DefaultSidecarProxyAddr = "127.0.0.1:5000/sidecar-proxy:latest"
+var DefaultSidecarProxyAddr = "127.0.0.1:5000/sidecar-proxy:1.0.0"
```
Comment (gallacher) [inline internal/web/client_install_handler.go:11]: Again, can we pull version out of the code?

File: deploy/install.go (hunk 1 above)
Comment (hoppea2) [reply deploy/install.go]: Done

File: docs/GETTING_STARTED_GUIDE.md (hunk 2 above)
Comment (hoppea2) [reply docs/GETTING_STARTED_GUIDE.md]: removed

File: docs/GETTING_STARTED_GUIDE.md (hunk 3 above)
Comment (hoppea2) [reply docs/GETTING_STARTED_GUIDE.md]: removed

File: docs/GETTING_STARTED_GUIDE.md (hunk 4 above)
Comment (hoppea2) [reply docs/GETTING_STARTED_GUIDE.md]: removed

File: docs/KARAVICTL_GUIDE.md (hunk 5 above)
Comment (hoppea2) [reply docs/KARAVICTL_GUIDE.md]: removed

File: internal/web/client_install_handler.go (hunk 6 above)
Comment (hoppea2) [reply internal/web/client_install_handler.go:11]: This value is always overwritten. We could leave it as "latest"

**Review by gallacher — DISMISSED** [review]

File: cmd/sidecar-proxy/main.go [hunk 7]
```diff
@@ -127,15 +127,6 @@ func (pi *ProxyInstance) Start(proxyHost, access, refresh string) error {
 			},
 		}
 	}
-	pi.rp.ModifyResponse = func(r *http.Response) error {
```
Comment (AronAtDell) [inline cmd/sidecar-proxy/main.go:130]: Why are these changes being made in this PR?

File: internal/web/testdata/storage_pool_instances.json [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+                "rel": "/api/StoragePool/relationship/VTree",
+                "href": "/api/instances/StoragePool::3df6df7600000001/relationships/VTree"
+            },
+            {
+                "rel": "/api/parent/relationship/protectionDomainId",
+                "href": "/api/instances/ProtectionDomain::75b661b400000000"
+            }
+        ]
+    }
+]
```
Comment (AronAtDell) [inline internal/web/testdata/storage_pool_instances.json:196]: Newline issue.

File: cmd/sidecar-proxy/main.go (hunk 7 above)
Comment (hoppea2) [reply cmd/sidecar-proxy/main.go:130]: This was a debug statement that was left erroneously as part of the powerscale feature.

**Review by gallacher — APPROVED** [review]
LGTM

**Review by atye — APPROVED** [review]

**Pull request merged**